spec/queries/company_users_collection_query_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CompanyUsersCollectionQuery do
+  describe '.call' do
+    subject(:call) { described_class.new(admin).call }
+
+    let(:company1) { create(:company) }
+    let(:user1) { create(:user, company: company1, name: 'userC') }
+    let(:user2) { create(:user, company: company1, name: 'userD') }
+
+    let(:company2) { create(:company) }
+    let(:user3) { create(:user, company: company2, name: 'userA') }
+    let(:user4) { create(:user, company: company2, name: 'userB') }
+
+    before do
+      user1
+      user2
+      user3
+      user4
```

Comment:
if you need the user to be created before running the test use let! instead https://rspec.info/documentation/3.10/rspec-core/RSpec/Core/MemoizedHelpers/ClassMethods.html#let!-instance_method